src/diffusers/models/unets/stable_diffusion_unet.py

```diff
@@ -0,0 +1,1306 @@
+# Copyright 2023 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from dataclasses import dataclass
+from typing import Any, Dict, Optional, Tuple, Union
+
+import torch
+import torch.nn as nn
+import torch.utils.checkpoint
+
+from ...configuration_utils import ConfigMixin, register_to_config
+from ...loaders import UNet2DConditionLoadersMixin
+from ...utils import (
+    USE_PEFT_BACKEND,
+    BaseOutput,
+    deprecate,
+    is_torch_version,
+    logging,
+    scale_lora_layers,
+    unscale_lora_layers,
+)
+from ...utils.torch_utils import apply_freeu
+from ..activations import get_activation
+from ..attention_processor import (
+    ADDED_KV_ATTENTION_PROCESSORS,
+    CROSS_ATTENTION_PROCESSORS,
+    Attention,
+    AttentionProcessor,
+    AttnAddedKVProcessor,
+    AttnProcessor,
+)
+from ..downsampling import Downsample2D
+from ..embeddings import (
+    TimestepEmbedding,
+    Timesteps,
+)
+from ..modeling_utils import ModelMixin
+from ..resnet import ResnetBlock2D
+from ..transformer_2d import Transformer2DModel
+from ..unet_2d_blocks import (
+    UNetMidBlock2DCrossAttn,
+)
+from ..upsampling import Upsample2D
+
+
+logger = logging.get_logger(__name__)  # pylint: disable=invalid-name
+
+
+class SDCrossAttnDownBlock2D(nn.Module):
```

Comment:
Do we want to keep these blocks in this file? I thought the higher-level UNet implementations would only collate lower-level blocks while them residing elsehwere.